digdag-standards/src/main/java/io/digdag/standards/operator/aws/EmrOperatorFactory.java

```diff
@@ -181,10 +181,37 @@ public TaskResult runTask()
             AWSCredentials credentials = credentials(tag);
 
             SecretProvider awsSecrets = context.getSecrets().getSecrets("aws");
+            SecretProvider s3Secrets = awsSecrets.getSecrets("s3");
+            SecretProvider emrSecrets = awsSecrets.getSecrets("emr");
+            SecretProvider kmsSecrets = awsSecrets.getSecrets("kms");
+
+            Optional<String> emrEndpoint = Aws.first(
+                    () -> emrSecrets.getSecretOptional("endpoint"),
+                    () -> params.getOptional("endpoint", String.class));
```

Comment:
Ah you are right - there is definitely an inconsistency. Maybe all endpoints should be set like so:

```
Aws.first(
  () -> emrSecrets.getSecretOptional("endpoint"),
  () -> awsSecrets.getSecretOptional("emr.endpoint"),
  () -> params.getOptional("emr.endpoint", String.class))
...
Aws.first(
   () -> s3Secrets.getSecretOptional("endpoint"),
   () -> awsSecrets.getSecretOptional("s3.endpoint"),
   () -> params.getOptional("s3.endpoint", String.class))
...
Aws.first(
  () -> kmsSecrets.getSecretOptional("endpoint"),
  () -> awsSecrets.getSecretOptional("kms.endpoint"),
  () -> params.getOptional("kms.endpoint", String.class));
```

What do you think?